report/report.tex

```diff
@@ -37,8 +37,30 @@
 \end{IEEEkeywords}
 
 \section{Introduction}
+Digital door locks are becoming more and more common, and as they become more common, the vulnerabilities they suffer from become more and more dangerous. While digital locks are more convenient by offering a keyless entry and exit, they are susceptible to a slew of different attacks, all intended to derive the static PIN that unlocks the digital lock.
+
 \subsection{Problems with traditional keypad entry systems}
 
+Assuming a 4 digit access code, simple wear and tear on the keypad means that the attacker only needs to try 4! (24) combinations at most to bruteforce the PIN, cutting down the original keyspace by almost 98\% (Figure 1).
+
+\begin{figure}[h!]
+  \includegraphics[width=\linewidth]{worn_lock.png}
+  \caption{Worn keypad exposes digits of PIN}
+  \label{fig:wornlock}
+\end{figure}
+
+More active attacks include shoulder surfing or installing inconspicuous pinhole cameras over the lock, revealing to the attacker exactly what the combination is if the victim is unaware of the attack. 
+
+Clearly, this is a security problem, one that hinges on the assumption of a static PIN code. Therefore, we seek to address this problem by dynamically and randomly generating temporary access codes for the user on demand, rather than having the lock accept a preset access code. In other words, our system will have the digital lock, by default, reject every input as invalid. Only when users request a new code through our mobile app will the lock accept the newly generated code.
+
+\subsection{Importance of Problem}
+It can be hard to quantify or estimate the value of goods protected by electronic keypad locks. We postulate that if the location is important enough to secure with an electronic keypad lock, then the user does not want the location compromised, and thus security is important to the user. 
```

Comment:
maybe mention that we have noticed these locks being used in places such as homes or commercial buildings, and even places with national security implications such as water treatment plants or electrical substations